bpf/kmesh/ads/include/route_config.h

```diff
@@ -334,6 +334,89 @@ static inline char *route_get_cluster(const Route__Route *route)
     return kmesh_get_ptr_val(_(route_act->cluster));
 }
 
+static inline __u32 lb_hash_based_header(unsigned char *header_name)
+{
+    __u32 hash = 0;
+    struct bpf_mem_ptr *msg_header = NULL;
+    char *msg_header_v = NULL;
+    __u32 msg_header_len = 0;
+    __u32 k;
+    __u32 c;
+    char msg_header_cp[KMESH_PER_HASH_POLICY_MSG_HEADER_LEN] = {'\0'};
+
+    if (!header_name)
+        return hash;
+    // when header name is not null, compute a hash value.
+    BPF_LOG(INFO, ROUTER_CONFIG, "Got a header name:%s\n", header_name);
+    msg_header = (struct bpf_mem_ptr *)bpf_get_msg_header_element(header_name);
+    if (!msg_header)
+        return hash;
+
+    msg_header_v = _(msg_header->ptr);
+    msg_header_len = _(msg_header->size);
+    if (!msg_header_len)
+        return hash;
+
+    BPF_LOG(INFO, ROUTER_CONFIG, "Got a header value len:%u\n", msg_header_len);
+    if (!bpf_strncpy(msg_header_cp, msg_header_len, msg_header_v))
+        return hash;
+    BPF_LOG(INFO, ROUTER_CONFIG, "Got a header value:%s\n", msg_header_cp);
+    hash = 5318;
+#pragma unroll
+    for (k = 0; k < KMESH_PER_HASH_POLICY_MSG_HEADER_LEN; k++) {
+        if (!msg_header_v || k >= msg_header_len)
+            break;
+        c = *((unsigned char *)msg_header_cp + k);
+        hash = ((hash << 5) + hash) + c;
+    }
+
+    return hash;
+}
+
+static inline int map_add_lb_hash(__u32 hash)
+{
+    int location = 0;
```

Comment:
It seems that key conflicts exist. different sockets use the same key, and records may be overwritten.

Reply:
This is a bug, will fix in next Pr.